Week01/Problem04/afilbrun_04.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jul 12 13:32:50 2018
+
+@author: afilbrun
+"""
+
+# Problem 4
+palindrome = []; 
+for first in range (100,1000):
+    for second in range (100,1000):
+        num = first * second;
+        if str(num) == str(num)[::-1]:
+            palindrome.append(num)
```

Comment:
If you were worried about the size of the list `palindrome` (like if we were searching between 100 and 100000000, for example), you could avoid storing all such palindromes by simply comparing to the previous one to return the maximum, i.e.,
```
if num > previous: 
    previous = num
```